ERROR! Unexpected Exception, this is probably a bug: 'AnsibleSequence' object has no attribute 'rfind'

#### SUMMARY
I'm trying to loop over a role with one variable being changed at each play but I get the following error message
```
ERROR! Unexpected Exception, this is probably a bug: 'AnsibleSequence' object has no attribute 'rfind'
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
include_role

##### ANSIBLE VERSION
```
ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/altiire/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.16 (default, Oct 10 2019, 22:02:15) [GCC 8.3.0]
```

##### CONFIGURATION
```
ALLOW_WORLD_READABLE_TMPFILES(/etc/ansible/ansible.cfg) = True
```

##### OS / ENVIRONMENT
OS Debian 10
Hosted on GCP


##### STEPS TO REPRODUCE
```yaml
- name: Update the deployments
  hosts: all
  remote_user: "{{ remote_user }}"

  vars_files:
    - vars/common.yml
  vars:
    - deployment_template_path: "{{ deployment_path }}/deployment_template"

  tasks:
    - include_role:
        name: deploy
        vars_from:
          - "./vars/deployment_{{ item }}.yml"
      vars:
        - current_deployment_path: "{{ deployment_path }}/{{ item }}"
        - deployment_name: "{{ item }}"
      loop: "{{ existing_deployments.stdout_lines }}"
```

##### EXPECTED RESULTS
run deploy role for each existing deployment


##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->

<!--- Paste verbatim command output between quotes -->
```paste below
ansible-playbook -vvv --connection=local --inventory 127.0.0.1, scale_down_es.yml -K --ask-vault-pass --extra-vars "swarm=false"
ansible-playbook 2.9.6
  config file = /etc/ansible/ansible.cfg                                             
  configured module search path = [u'/home/altiire/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible               
  executable location = /usr/bin/ansible-playbook         
  python version = 2.7.16 (default, Oct 10 2019, 22:02:15) [GCC 8.3.0]                       
Using /etc/ansible/ansible.cfg as config file
BECOME password:                                                                                      
Vault password:   
Parsed 127.0.0.1, inventory source with host_list plugin                                     
Read vars_file 'vars/common.yml'                         
Read vars_file 'vars/common.yml'                               
statically imported: /home/altiire/deployer/roles/scale-down-es/tasks/prepare.yml
Read vars_file 'vars/common.yml'                                 
statically imported: /home/altiire/deployer/roles/scale-down-es/tasks/scale_down.yml
Read vars_file 'vars/common.yml'                                                                                                                                                                                   
Read vars_file 'vars/common.yml'                                        
statically imported: /home/altiire/deployer/roles/prepare/tasks/git.yml
Read vars_file 'vars/common.yml'        
statically imported: /home/altiire/deployer/roles/prepare/tasks/backend.yml                                                                                                                                        
Read vars_file 'vars/common.yml'                                           
statically imported: /home/altiire/deployer/roles/prepare/tasks/frontend.yml
Read vars_file 'vars/common.yml'                                      
statically imported: /home/altiire/deployer/roles/prepare/tasks/sftp.yml                 
Read vars_file 'vars/common.yml'
statically imported: /home/altiire/deployer/roles/prepare/tasks/elasticsearch.yml
Read vars_file 'vars/common.yml'
statically imported: /home/altiire/deployer/roles/prepare/tasks/docker.yml
Read vars_file 'vars/common.yml'
statically imported: /home/altiire/deployer/roles/prepare/tasks/reconnect.yml
ERROR! Unexpected Exception, this is probably a bug: 'AnsibleSequence' object has no attribute 'rfind'
the full traceback was:

Traceback (most recent call last):
  File "/usr/bin/ansible-playbook", line 123, in <module>
    exit_code = cli.run()
  File "/usr/lib/python2.7/dist-packages/ansible/cli/playbook.py", line 127, in run
    results = pbex.run()
  File "/usr/lib/python2.7/dist-packages/ansible/executor/playbook_executor.py", line 91, in run
    pb = Playbook.load(playbook_path, variable_manager=self._variable_manager, loader=self._loader)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/__init__.py", line 51, in load
    pb._load_playbook_data(file_name=file_name, variable_manager=variable_manager)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/__init__.py", line 103, in _load_playbook_data
    entry_obj = Play.load(entry, variable_manager=variable_manager, loader=self._loader, vars=vars)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/play.py", line 116, in load
    return p.load_data(data, variable_manager=variable_manager, loader=loader)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/base.py", line 235, in load_data
    self._attributes[target_name] = method(name, ds[name])
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/play.py", line 147, in _load_tasks
    return load_list_of_blocks(ds=ds, play=self, variable_manager=self._variable_manager, loader=self._loader)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/helpers.py", line 78, in load_list_of_blocks
    loader=loader,
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/block.py", line 94, in load
    return b.load_data(data, variable_manager=variable_manager, loader=loader)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/base.py", line 235, in load_data
    self._attributes[target_name] = method(name, ds[name])
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/block.py", line 130, in _load_block
    use_handlers=self._use_handlers,
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/helpers.py", line 325, in load_list_of_tasks
    loader=loader,
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/role_include.py", line 140, in load
    ir._from_files[from_key] = basename(ir.args.get(key))
  File "/usr/lib/python2.7/posixpath.py", line 114, in basename
    i = p.rfind('/') + 1
AttributeError: 'AnsibleSequence' object has no attribute 'rfind'
make: *** [Makefile:45: scale-down-es] Error 250
```

--- a/changelogs/fragments/68515_include_role_vars_from.yml
+++ b/changelogs/fragments/68515_include_role_vars_from.yml
@@ -1,0 +1,2 @@
+bugfixes:
+- Strictly check string datatype for 'tasks_from', 'vars_from', 'defaults_from', and 'handlers_from' in include_role (https://github.com/ansible/ansible/issues/68515).

--- a/lib/ansible/playbook/role_include.py
+++ b/lib/ansible/playbook/role_include.py
@@ -27,6 +27,7 @@ from ansible.playbook.task_include import TaskInclude
 from ansible.playbook.role import Role
 from ansible.playbook.role.include import RoleInclude
 from ansible.utils.display import Display
+from ansible.module_utils.six import string_types
 
 __all__ = ['IncludeRole']
 
@@ -137,7 +138,10 @@ class IncludeRole(TaskInclude):
         # build options for role includes
         for key in my_arg_names.intersection(IncludeRole.FROM_ARGS):
             from_key = key.replace('_from', '')
-            ir._from_files[from_key] = basename(ir.args.get(key))
+            args_value = ir.args.get(key)
+            if not isinstance(args_value, string_types):
+                raise AnsibleParserError('Expected a string for %s but got %s instead' % (key, type(args_value)))
+            ir._from_files[from_key] = basename(args_value)
 
         apply_attrs = ir.args.get('apply', {})
         if apply_attrs and ir.action != 'include_role':

--- a/test/integration/targets/include_import/role/test_include_role.yml
+++ b/test/integration/targets/include_import/role/test_include_role.yml
@@ -153,13 +153,13 @@
       include_role:
         name: "{{ host_var_role_name }}"
 
-    - name: assert that host varible for role name calls 2 diff roles
+    - name: assert that host variable for role name calls 2 diff roles
       assert:
         that:
           - _role2_result is not none
       when: inventory_hostname == 'testhost2'
 
-    - name: assert that host varible for role name calls 2 diff roles
+    - name: assert that host variable for role name calls 2 diff roles
       assert:
         that:
           - _role3_result is not none

--- a/test/integration/targets/include_import/role/test_include_role_vars_from.yml
+++ b/test/integration/targets/include_import/role/test_include_role_vars_from.yml
@@ -1,0 +1,10 @@
+- name: Test include_role vars_from
+  hosts: testhost
+  vars:
+    role_name: role1
+  tasks:
+    - name: Test vars_from
+      include_role:
+        name: role1
+        vars_from:
+        - vars_1.yml

--- a/test/integration/targets/include_import/runme.sh
+++ b/test/integration/targets/include_import/runme.sh
@@ -42,6 +42,9 @@ ANSIBLE_STRATEGY='free' ansible-playbook tasks/test_include_tasks_tags.yml -i in
 ANSIBLE_STRATEGY='linear' ansible-playbook role/test_include_role.yml -i inventory "$@"
 ANSIBLE_STRATEGY='free' ansible-playbook role/test_include_role.yml -i inventory "$@"
 
+# https://github.com/ansible/ansible/issues/68515
+ansible-playbook -v role/test_include_role_vars_from.yml 2>&1 | tee test_include_role_vars_from.out
+test "$(grep -E -c 'Expected a string for vars_from but got' test_include_role_vars_from.out)" = 1
 
 ## Max Recursion Depth
 # https://github.com/ansible/ansible/issues/23609

